### PR TITLE
feat: weather api

### PR DESCRIPTION
**Pull request by muskaanshah**
features:
updating weather based on user's current location,
If user wants to search and update for a specific location, that can be done as well by hovering on the temperature div (top right corner)

### CODE REVIEW

File: src/components/Weather/Weather.jsx [hunk 1]
```diff
@@ -1,8 +1,63 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+const APIKEY = "40d6f448e392001b5598241621cd7b46";
+
 function Weather() {
+	const [city, setCity] = useState("");
```
Comment (xsaurabhsuthar) [inline src/components/Weather/Weather.jsx:7]: suggestion: you can extract the logic in the WeatherContainer.js using logic-presentation layer design.

**Review by xsaurabhsuthar — APPROVED** [review]
- Clean code 
- Good folder structure.
Good to merge 🚀🚀🚀🚀

File: src/components/Weather/Weather.jsx [hunk 2]
```diff
@@ -1,8 +1,63 @@
...
+	}, [cityValue]);
+
+	const weatherAPI = async (lat, lon) => {
+		let API = "";
+		cityValue.length <= 0
+			? (API = `https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&exclude={part}&appid=${APIKEY}`)
+			: (API = `https://api.openweathermap.org/data/2.5/weather?q=${cityValue}&appid=${APIKEY}`);
+		try {
+			const res = await axios.get(API);
+			console.log(res);
```
Comment (Pratik1005) [inline src/components/Weather/Weather.jsx:19]: If you are done with testing you can remove console log

**Review by Pratik1005 — APPROVED** [review]
Code looks good 👍

**Pull request merged**